modules/dnn/src/layers/reduce2_layer.cpp

```diff
@@ -0,0 +1,601 @@
+            case ReduceType::MEAN:        ReduceInvoker<ReduceMean<T>>::run(std::forward<Args>(args)...);      break;
+            case ReduceType::SUM:         ReduceInvoker<ReduceSum<T>>::run(std::forward<Args>(args)...);       break;
+            case ReduceType::L1:          ReduceInvoker<ReduceL1<T>>::run(std::forward<Args>(args)...);        break;
+            case ReduceType::L2:          ReduceInvoker<ReduceL2<T>>::run(std::forward<Args>(args)...);        break;
+            case ReduceType::PROD:        ReduceInvoker<ReduceProd<T>>::run(std::forward<Args>(args)...);      break;
+            case ReduceType::SUM_SQUARE:  ReduceInvoker<ReduceSumSquare<T>>::run(std::forward<Args>(args)...); break;
+            case ReduceType::LOG_SUM:     ReduceInvoker<ReduceLogSum<T>>::run(std::forward<Args>(args)...);    break;
+            case ReduceType::LOG_SUM_EXP: ReduceInvoker<ReduceLogSumExp<T>>::run(std::forward<Args>(args)...); break;
+            default: CV_Error(Error::StsBadArg, "DNN/Reduce: Unsupported operation.");
+        }
+    }
+
+    template <typename... Args>
+    inline void typeDispatch(const int type, Args&&... args) {
+        switch (type) {
+            case CV_8U:  opDispatch<uint8_t>(std::forward<Args>(args)...); break;
+            case CV_8S:  opDispatch<int8_t>(std::forward<Args>(args)...);  break;
+            case CV_32S: opDispatch<int32_t>(std::forward<Args>(args)...); break;
+            case CV_64S: opDispatch<int64_t>(std::forward<Args>(args)...); break;
+            case CV_32F: opDispatch<float>(std::forward<Args>(args)...);   break;
+            default: CV_Error(cv::Error::BadDepth, "DNN/Reduce: Unsupported type.");
+        }
+    }
+
+#ifdef HAVE_CANN
+    virtual Ptr<BackendNode> initCann(const std::vector<Ptr<BackendWrapper> > &inputs,
+                                      const std::vector<Ptr<BackendWrapper> > &outputs,
+                                      const std::vector<Ptr<BackendNode> >& nodes) CV_OVERRIDE
+    {
+        CV_Check(!axes_from_params.empty(), "DNN/CANN: Reduce2 needs constant axes to build CANN operators");
+
+        auto input_node = nodes[0].dynamicCast<CannBackendNode>()->getOp();
+        auto input_wrapper = inputs[0].dynamicCast<CannBackendWrapper>();
+        auto input_desc = input_wrapper->getTensorDesc();
+
+        std::vector<int> axes_shape{(int)axes_from_params.size()};
+        Mat axes_mat(axes_shape, CV_32S, axes_from_params.data());
+        auto axes_node = std::make_shared<CannConstOp>(axes_mat.data, axes_mat.type(), axes_shape, cv::format("%s_axes", name.c_str()));
+        auto axes_desc = axes_node->getTensorDesc();
+
+        auto output_desc = std::make_shared<ge::TensorDesc>(ge::Shape(), ge::FORMAT_NCHW, ge::DT_FLOAT);
+
+        std::shared_ptr<ge::Operator> reduce_op = nullptr;
+        switch (reduce_type)
+        {
+#define BUILD_CANN_REDUCE_OP(op_type, class_name, op_name)                         \
```

Comment:
please, remove CANN branch to keep the new implementation clean for now. We will add CANN back later